core/src/Dscp/Core/Foundation/Witness.hs

```diff
@@ -190,10 +191,9 @@ toPtxId = hash
 
 -- | Incoming publication message; transaction payload.
 data Publication = Publication
-    { pPrivateBlockHash  :: PrivateHeaderHash
-    , pPreviousBlockHash :: Maybe PrivateHeaderHash
-    }
-    deriving (Eq, Ord, Show, Generic)
+    { pPrivateHeader  :: PrivateBlockHeader
+    , pPreviousHeader :: Maybe PrivateBlockHeader
```

Comment:
Same question. Why do we need to send previous block header, if we can send hash of previous block header?

Reply:
Also, hash of previous block header is already contained in `PrivateBlockHeader` :man_shrugging: